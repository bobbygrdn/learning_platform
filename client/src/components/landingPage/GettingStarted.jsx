import React from 'react'

export default function GettingStarted() {
    // TODO: Add src urls for each arrow and each step
    return (
        <>
            <img className="step" src="" alt="Step 1" />
            <img className="arrow" src="" alt="Arrow" />
            <img className="step" src="" alt="Step 2" />
            <img className="arrow" src="" alt="Arrow" />
            <img className="step" src="" alt="Step 3" />
            <img className="arrow" src="" alt="Arrow" />
            <img className="step" src="" alt="Step 4" />

        </>
    )
}
