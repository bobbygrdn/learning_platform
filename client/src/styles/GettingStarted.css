.gettingStarted {
    display: grid;
    grid-template-rows: repeat(3, 1fr);
    grid-template-columns: repeat(3, 1fr);
    grid-gap: 1em;
    margin-bottom: 5em;
    padding: 0 1em;
}

.steps {
    font-size: 30px;
    margin-bottom: 2em;
    pointer-events: none;
}

.step {
    width: 12em;
    border: .2em solid #677D8C;
    border-radius: .5em;
}

.arrow {
    width: 10em;
}

.arrow2 {
    transform: rotate(90deg);
}

@media screen and (max-width: 600px) {
    .gettingStarted {
        grid-template-rows: repeat(3, 1fr);
        grid-template-columns: repeat(3, 1fr);
    }

    .steps {
        text-align: center;
        font-size: 24px;
        margin-bottom: 2em;
        padding: 0 1em;
    }

    .step {
        width: 6em;
    }

    .arrow {
        width: 6em;
    }
}