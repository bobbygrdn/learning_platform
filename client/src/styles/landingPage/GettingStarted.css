.gettingStarted {
  display: grid;
  grid-template-rows: repeat(3, 1fr);
  grid-template-columns: repeat(3, 1fr);
  grid-gap: 1em;
  margin-bottom: 5em;
  padding: 0 1em;
}

.steps {
  font-size: 30px;
  margin-bottom: 2em;
  pointer-events: none;
}

.step {
  width: 12em;
  border-radius: 0.5em;
  box-shadow: 0px 0px 20px #677d8c;
}

.arrow {
  width: 10em;
}

.arrow1,
.arrow4 {
  margin-top: 4em;
  padding-left: 2em;
}

.arrow2 {
  margin-top: 4em;
  transform: rotate(90deg);
}

.arrow3 {
  margin-top: 4em;
}

@media screen and (max-width: 600px) {
  .gettingStarted {
    grid-template-rows: repeat(3, 1fr);
    grid-template-columns: repeat(3, 1fr);
  }

  .steps {
    text-align: center;
    font-size: 24px;
    margin-bottom: 2em;
    padding: 0 1em;
  }

  .step {
    width: 6em;
  }

  .arrow {
    width: 6em;
  }

  .arrow1,
  .arrow4 {
    margin-top: 2em;
    padding-left: 0;
  }

  .arrow2 {
    margin-top: 1.5em;
  }

  .arrow3 {
    margin-top: 1.5em;
  }
}
