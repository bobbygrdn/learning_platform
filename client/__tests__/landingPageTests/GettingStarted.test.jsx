import React from 'react';
import { render, screen } from '@testing-library/react';
import GettingStarted from '../../src/components/landingPage/GettingStarted';

it('renders without crashing', () => {
    const { container } = render(<GettingStarted />);
    expect(container).toMatchSnapshot();
})

it('should render paragraph with steps class name', () => {
    render(<GettingStarted />);
    expect(screen.getAllByTestId('steps')).toHaveLength(1);
})

it('should render five image elements with step class name', () => {
    render(<GettingStarted />);
    expect(screen.getAllByTestId('step')).toHaveLength(5);
})

it('should render four images of arrow icon with arrow class name', () => {
    render(<GettingStarted />);
    expect(screen.getAllByTestId('arrow')).toHaveLength(4);
})