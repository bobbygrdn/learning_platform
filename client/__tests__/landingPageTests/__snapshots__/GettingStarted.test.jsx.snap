// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<div>
  <p
    class="steps"
    data-testid="steps"
  >
    Here is a step by step guide to get you started!
  </p>
  <div
    class="gettingStarted"
  >
    <img
      alt="Step 1"
      class="step"
      data-testid="step"
      src="/resources/male_meijin.jpg"
    />
    <img
      alt="Arrow"
      class="arrow arrow1"
      data-testid="arrow"
      src="/resources/arrow_right.png"
    />
    <img
      alt="Step 2"
      class="step"
      data-testid="step"
      src="/resources/male_meijin.jpg"
    />
    <img
      alt="Arrow"
      class="arrow arrow2"
      data-testid="arrow"
      src="/resources/curved_arrow_right.png"
    />
    <img
      alt="Step 3"
      class="step"
      data-testid="step"
      src="/resources/male_meijin.jpg"
    />
    <img
      alt="Arrow"
      class="arrow arrow3"
      data-testid="arrow"
      src="/resources/curved_arrow_left.png"
    />
    <img
      alt="Step 4"
      class="step"
      data-testid="step"
      src="/resources/male_meijin.jpg"
    />
    <img
      alt="Arrow"
      class="arrow arrow4"
      data-testid="arrow"
      src="/resources/arrow_right.png"
    />
    <img
      alt="Step 5"
      class="step"
      data-testid="step"
      src="/resources/male_meijin.jpg"
    />
  </div>
</div>
`;
